### DISCUSSION TITLE
Intra-word emphasis

### DISCUSSION THREAD

**Discussion by ahamelers** [Milkdown/milkdown#1157] [Q&A] [answered]
Is it possible at all to turn off intra-word emphasis, and more particularly, the escaping of underscores in snake_case_file_names when extracting markdown from the editor?

My use case is to allow people to create README.md files that include many, many references to filenames. The insertion of escape characters inside words is not helpful when sharing those README files as plain text.

**Comment by Saul-Mirone** [comment]
Yes, you can configure the inlineSyncPlugin, the [default config is defined here](https://github.com/Milkdown/milkdown/blob/main/packages/preset-commonmark/src/plugin/inline-sync-plugin/config.ts).

Maybe this is what you want:
```ts
Editor.make()
 .config(ctx=> {
    ctx.update(inlineSyncConfig.key, (defaultConfig) => {
      return {
        ...defaultConfig,
        movePlaceholder: (placeholderToMove: string, text: string) => {
          const symbolsNeedToMove = ['*']
          function swap(text: string, first: number, last: number) {
            const arr = text.split('')
            const temp = arr[first]
            if (arr[first] && arr[last]) {
              arr[first] = arr[last] as string
              arr[last] = temp as string
            }
            return arr.join('').toString()
          }

      
          let index = text.indexOf(placeholderToMove)
          while (symbolsNeedToMove.includes(text[index - 1] || '') && symbolsNeedToMove.includes(text[index + 1] || '')) {
            text = swap(text, index, index + 1)
            index = index + 1
          }
      
          return text
        },
      }
    })
 })
```

With this method, you'll prevent the `_` character to be transformed. However, this will break the feature that when users type `_something_`, the word won't be transformed into italic style.

**Reply by ahamelers** [reply]
Many markdown parsers have an intra-word-emphasis setting, such that they would convert `_something_` to italic, but would not convert "word" in  `intra_word_emphasis` to italic. Is that possible in milkdown if I fiddle with regexes enough, or is turning off `_` emphasis marking entirely the only option?

**Reply by ahamelers** [reply]
With the code above `_` still all get escaped in the markdown output. I think it's a later process, not the inline sync, that I need to modify.

**Reply by Saul-Mirone** [reply, marked as answer]
I think milkdown also won't parse `intra_word_emphasis` to italic. If you mean the output file will transform it into `intra\_word\_emphasis`. I think it's the behavior of remark. Maybe you can find a way to configure the `remark-stringify` to change the behavior.

https://stackblitz.com/edit/stackblitz-starters-7orxg2?file=index.js

**Reply by ahamelers** [reply]
The markdown I am referring to is the output of the plugin-listener and other markdown retrieval functions like
```
const view = ctx.get(editorViewCtx);
const serializer = ctx.get(serializerCtx);
const markdown = serializer(view.state.doc);
```
So that seems to be the serializer. Is the serializer controlled by remark?
